### ISSUE TITLE
Don't misuse goto to imitate priority queues

### ISSUE THREAD

**Issue by Al2Klimov** [Icinga/icinga2#8548]
This one...

https://github.com/Icinga/icinga2/blob/826f999c47270f344b93ac6f30bed2abf5fbd8e2/lib/icingadb/redisconnection.hpp#L167-L168

... orders its items by priority (`std::map<QueryPriority,`) and then by insertion order (`std::queue<WriteQueueItem>`).

Replace it with a `std::priority_queue<WriteQueueItem>`. Adjust `WriteQueueItem` and the usages of `Writes` as needed.

**Comment by Al2Klimov** [comment]
My intention was to make it more clear, not [more complicated](https://github.com/Icinga/icinga2/pull/8557#issuecomment-760753038).

**Comment by julianbrost** [comment]
@Al2Klimov Can you explain why you labeled this as bug? As far as I can tell the current code works as expected. So if there's no reason to consider this a bug, I'd close this as there was an attempt that didn't work out.

**Comment by Al2Klimov** [comment]
No, I labeled it bug+core/quality because to me it's a bug in our code quality.

**Comment by Al2Klimov** [comment]
Wait... why close?

**Comment by julianbrost** [comment]
That's not a bug for me. Sure, avoiding goto would be nice, but it's not like this place in the code is a terribly convoluted goto hell that needs to be fixed because nobody understands what's actually going on there.
